Automattic/wp-calypso
# Update/inline help data flow

#### Changes proposed in this Pull Request

This PR changes the way the `<InlineHelp />` requests the data. Right now, it performs the requests imperatively, executing the `setInlineHelpQuerySearch()` action from the UI. It should be handled by `<QueryInlineHelpSearch />`, the data query component designed for this purpose.
For this, I've added an action used to set the query string from the component/UI. This property is passed to the data query component to finally perform the requests.

#### Testing instructions

Fist at all, check that either `<InlineHelp />` as well as `<HelpSearch />` works in the same way that it rightly works on production.

You can take a look at how the app dispatches the actions while you type in the query string input.

<img width="702" alt="Screen Shot 2020-06-29 at 5 59 21 PM" src="https://user-images.githubusercontent.com/77539/86055708-95a57080-ba32-11ea-9648-bf2e7ab5076c.png">

<img width="666" alt="Screen Shot 2020-06-29 at 5 59 52 PM" src="https://user-images.githubusercontent.com/77539/86055705-950cda00-ba32-11ea-8500-8fb4915153a6.png">

<img width="873" alt="Screen Shot 2020-06-29 at 6 00 20 PM" src="https://user-images.githubusercontent.com/77539/86055695-9211e980-ba32-11ea-9235-988938126d70.png">

Fixes #43624
